### PR TITLE
add zemu integration test

### PR DESCRIPTION
**Pull request by linfeng-crypto**
add zemu integration test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "@babel/helper-builder-react-jsx": "^7.10.4",
     "@babel/helper-builder-react-jsx-experimental": "^7.12.11",
     "@cosmjs/stargate": "^0.24.0-alpha.22",
-    "@crypto-com/chain-jslib": "^0.0.10",
+    "@crypto-com/chain-jslib": "linfeng-crypto/chain-jslib#dev2",
```
Comment (leejw51crypto) [inline package.json:16]: could you submit the modification to chain-jslib?

File: tests/src/test.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+import { LedgerSigner } from '../../src/service/signers/LedgerSigner';
+import { ISignerProvider } from '../../src/service/signers/SignerProvider';
+import { LedgerTransactionSigner } from '../../src/service/signers/LedgerTransactionSigner';
+import {CustomDevNet} from '../../src/config/StaticConfig';
+import { Bytes } from '@crypto-com/chain-jslib/lib/dist/utils/bytes/bytes';
+import {NodeRpcService} from "../../src/service/rpc/NodeRpcService";
+const { exec } = require("child_process");
+import chai from "chai";
+
+const APP_PATH = path.resolve(`./app/bin/app.elf`);
```
Comment (leejw51crypto) [inline tests/src/test.ts:13]: how about adding fullpath as an example?

File: tests/src/test.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    const cmd2 = "chain-maind keys add validator1 --keyring-backend test";
+    const cmd3 = "chain-maind keys add hw  --ledger --keyring-backend test";
+    const cmd4 = "chain-maind add-genesis-account $(chain-maind keys show validator1 -a --keyring-backend test) 200000000cro";
+    const cmd5 = "chain-maind add-genesis-account $(chain-maind keys show hw -a --keyring-backend test) 100000000cro";
+    const cmd6 = "chain-maind gentx validator1 100000000cro --keyring-backend test  --chain-id test";
+    const cmd7 = "chain-maind collect-gentxs";
+    const commands = [cmd1, cmd2, cmd3, cmd4, cmd5, cmd6, cmd7];
+    let i = 0;
+    for (let cmd of commands) {
+        await Zemu.default.sleep(3000);
```
Comment (leejw51crypto) [inline tests/src/test.ts:70]: const ZEMU_WAIT_SECONDS=3000;
and using const?

File: tests/src/test.ts [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+          console.log("double click");
+          await this.sim.clickBoth();
+        }
+    }
+    console.log("stop grpc server");
+    this.sim.stopgrpcServer();
+    console.log("start chain-maind");
+    exec("sed -i.bak 's/cors_allowed_origins = \\[\\]/cors_allowed_origins = [\"*\"]/g' $HOME/.chain-maind/config/config.toml");
+    await Zemu.default.sleep(3000);
+    // spawn("chain-maind", ["start", "--home", "$HOME/.chain-maind"]);
```
Comment (leejw51crypto) [inline tests/src/test.ts:90]: remove comment

**Review by leejw51crypto — DISMISSED** [review]
lgtm

**Comment by eddycpc** [comment]
Any gif or video of how the test is performing when running?  Since it's a UI integration test

**Comment by linfeng-crypto** [comment]
> Any gif or video of how the test is performing when running? Since it's a UI integration test

It's only run at the backend without UI and only test the core function of ledger part like get public key/get address/sign tx, because it needs to start the ledger simulator in a docker, so I think it's not a UI integration test, but you can see the test result from the actions: https://github.com/crypto-com/chain-desktop-wallet/actions

**Review by eddycpc — DISMISSED** [review]

**Review by eddycpc — APPROVED** [review]

**Comment by eddycpc** [comment]
@linfeng-crypto can you update the branch to the latest?

**Pull request merged**